Tasks/Src/RTOS_BPS.c

```diff
@@ -25,6 +25,20 @@ void RTOS_BPS_MutexPost(BPS_OS_MUTEX* mutex, BPS_OS_OPT options) {
     OSMutexPost(mutex, options, &err);
     assertOSError(err);
 }
+
+/**
+ * Note that the incoming BPS_OS_QUEUE queue !!MUST!! be created already.
+ * @brief   Puts a message into the RTOS Queue (i.e., POST something into the Queue). https://docs.silabs.com/micrium/latest/micrium-kernel-api/03-kernel-message-queue-api
+ * @param   *queue - pointer to the Queue to write a message to
+ * @param   *msg - void pointer to the message you are going to send
+ * @param   size - payload size of *msg (this changes in FreeRTOS)
+ * @return  none
+ */
+void RTOS_BPS_QueuePut(BPS_OS_QUEUE* queue, void* msg, BPS_OS_MSG_SIZE size) {
+    BPS_OS_ERR err;
+    OSQPost(queue, msg, size, OS_OPT_POST_FIFO, &err);
```

Comment:
is support for stack/LIFO planned?

Reply:
OS_OPT_POST_FIFO is stuff built into Micrium. If you mean using Queues overall, I think it might be used when we transition to FreeRTOS but I think Manthan would have a better idea about the broader scope of that. The wrapper exists for now as a more generic call for Queues and once we decide fully on when and how to use them it'll get some more updates to it.

Reply:
We don't use this function anywhere in the BPS code. I would recommend looking at fifo.h and CAN_Queue.c. Don't change it yet though, because I'm not too sure why we aren't using this function.

Reply:
I thought we don't use micrium queues anywhere and instead use fifo.h + semaphore

Reply:
Yea, I remember there being a reason why, but I don't remember what it is. I would assume the queues are thread safe. fifo.h + semaphore should probably be used in this function.